pkg/virt-handler/vm_test.go

```diff
@@ -955,16 +956,22 @@ var _ = Describe("VirtualMachineInstance", func() {
 			))
 
 			By("unpausing domain")
+			domain = domain.DeepCopy()
 			domain.Status.Status = api.Running
 			domain.Status.Reason = ""
 
-			vmiFeeder.Add(vmi)
+			addVMI(updatedVMI.DeepCopy())
 			addDomain(domain)
 
-			client.EXPECT().SyncVirtualMachine(vmi, gomock.Any())
+			client.EXPECT().SyncVirtualMachine(updatedVMI, gomock.Any())
 			mockHotplugVolumeMounter.EXPECT().Unmount(gomock.Any(), mockCgroupManager).Return(nil)
 			mockHotplugVolumeMounter.EXPECT().Mount(gomock.Any(), mockCgroupManager).Return(nil)
 
+			Expect(controller.domainStore.List()).To(ConsistOf(domain))
+			// TODO split this test
+			key, err := virtcontroller.KeyFunc(domain)
+			Expect(err).To(Not(HaveOccurred()))
+			controller.vmiExpectations.SetExpectations(key, 0, 0)
```

Comment:
Cam you explain this? I am a little bit lost

Reply:
An update is called in the previous sync, therefore here we need to reset the expectations. Previously the informers would do it.

Reply:
+1 for the double sync.
Thank you for the explanation